src/transformer/descriptor/method/bodyReturnType.ts

```diff
@@ -1,23 +1,26 @@
 import * as ts from 'typescript';
 import { Scope } from '../../scope/scope';
 import { GetDescriptor } from '../descriptor';
-import { GetNullDescriptor } from '../null/null';
 
-export function GetReturnTypeFromBody(node: ts.ArrowFunction | ts.FunctionExpression | ts.MethodDeclaration | ts.FunctionDeclaration, scope: Scope): ts.Expression {
-  let returnValue: ts.Expression;
+export function GetReturnTypeFromBodyDescriptor(node: ts.ArrowFunction | ts.FunctionExpression | ts.MethodDeclaration | ts.FunctionDeclaration, scope: Scope): ts.Expression {
+  return GetDescriptor(GetReturnNodeFromBody(node), scope);
+}
+
+export function GetReturnNodeFromBody(node: ts.ArrowFunction | ts.FunctionExpression | ts.MethodDeclaration | ts.FunctionDeclaration): ts.Node {
+  let returnValue: ts.Node;
 
   const functionBody: ts.FunctionBody = node.body as ts.FunctionBody;
```

Comment:
Because you are touching this file.. could we change this casting adding the type guard function call?? 

In this case I think it will be if (!ts.isArrowFunction(node))

Reply:
I did something slightly different but I think you will like it